packages/create-sitecore-jss/src/templates/nextjs/src/ServerError.tsx

```diff
@@ -0,0 +1,19 @@
+import Head from 'next/head';
+
+/**
+ * Rendered in case if we have 404 error
+ */
+const ServerError = (): JSX.Element => (
```

Comment:
It's true, I think this should just be moved to `500.tsx`. We have it split out with not found (`NotFound.tsx`) so that we can re-use between `404.tsx` and `[[...path]].tsx` but we don't have re-use here.

In general, it's my understanding that if you have `500.tsx` this will override `_error.tsx` (for server errors). I think it makes sense to have `_error.tsx` in base (like we have it), and `500.tsx` in sxa add-on. So, we'll have:
- `nextjs`:  `_error.tsx` > dynamic (can display error message), works for both client-side and server-side
- `nextjs-sxa`: `pages/500.tsx` > static (based on custom error page), overrides `_error.tsx` for server-side errors